src/platform/Zephyr/ConfigurationManagerImpl.cpp

```diff
@@ -204,9 +202,7 @@ void ConfigurationManagerImpl::DoFactoryReset(intptr_t arg)
     ThreadStackMgr().ErasePersistentInfo();
 #endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD
 
-#if CONFIG_REBOOT
```

Comment:
nit: I guess this change isn't that necessary since we will probably need to support some custom reboot mechanism (to send the shutdown event in any case).